libs/pilight/config/devices.c

```diff
@@ -308,11 +314,17 @@ int devices_update(char *protoname, JsonNode *json, enum origin_t origin, JsonNo
 								   && (opt->conftype == DEVICES_VALUE || opt->conftype == DEVICES_OPTIONAL)
 								   && opt->argtype == OPTION_HAS_VALUE) {
 									int upd_value = 1;
-									memset(vstring_, '\0', sizeof(vstring_));
 									vnumber_ = -1;
 									vdecimals_ = 0;
 									struct JsonNode *jtmp = NULL;
 									if(json_find_string(message, opt->name, &stmp) == 0) {
+										if((strlen(stmp) + 1) > vsize) {
```

Comment:
You didn't enclose this in a hook above.

Reply:
Yes indeed, that's not consistent.